pkg/storage/fingerprint_writer.go

```diff
@@ -180,3 +182,73 @@ func (f *fingerprintWriter) stripTenantPrefix(key []byte) []byte {
 	}
 	return remainder
 }
+
+// FingerprintRangekeys iterates over the provided SSTs, that are expected to
+// contain only rangekeys, and maintains a XOR aggregate of each rangekey's
+// fingerprint.
+func FingerprintRangekeys(
+	ctx context.Context, cs *cluster.Settings, opts MVCCExportFingerprintOptions, ssts [][]byte,
+) (uint64, error) {
+	ctx, sp := tracing.ChildSpan(ctx, "storage.FingerprintRangekeys")
+	defer sp.Finish()
+	_ = ctx // ctx is currently unused, but this new ctx should be used below in the future.
+
+	iterOpts := IterOptions{
+		KeyTypes:   IterKeyTypePointsAndRanges,
+		LowerBound: keys.LocalMax,
+		UpperBound: keys.MaxKey,
+	}
+	var fingerprint uint64
+	iter, err := NewMultiMemSSTIterator(ssts, false, iterOpts)
+	if err != nil {
+		return fingerprint, err
+	}
+	defer iter.Close()
+
+	destFile := &MemFile{}
+	fw := makeFingerprintWriter(ctx, fnv.New64(), cs, destFile, opts)
+	defer fw.Close()
+	fingerprintRangeKey := func(bounds roachpb.Span, versions MVCCRangeKeyVersions) (uint64, error) {
+		defer fw.hasher.Reset()
+		if err := fw.hashKey(bounds.Key); err != nil {
+			return 0, err
+		}
+		if err := fw.hashKey(bounds.EndKey); err != nil {
+			return 0, err
+		}
+		for _, v := range versions {
+			fw.timestampBuf = EncodeMVCCTimestampToBuf(fw.timestampBuf, v.Timestamp)
+			if err := fw.hash(fw.timestampBuf); err != nil {
+				return 0, err
+			}
+			if err := fw.hashValue(v.Value); err != nil {
```

Comment:
Good catch!

Reply:
nice catch indeed, i think we already have the test you outlined for point keys over here - https://github.com/cockroachdb/cockroach/blob/d316af081175e139273c95fedc20f7dcc5ed1606/pkg/storage/testdata/mvcc_histories/export_fingerprint_tenant#L11

Reply:
Now that we have `FingerprintRangekeys` I think I can teach TestMVCCHistories to also fingerprint rangekeys instead of printing out the rangekeys. Let me try that.